prestodb/dbapi.py

```diff
@@ -175,7 +181,9 @@ def execute(self, operation, params=None):
         )
 
         self._query = prestodb.client.PrestoQuery(request, sql=operation)
-        return self._query.execute()
+        result = self._query.execute()
+        self._iterator = iter(result)
```

Comment:
`PrestoResult` is an `iterable`; while `next` function (used in `fetchone`) can only apply on `iterator`. Convert it to `iterator`